### PR TITLE
Build pre-confirm dialog/modal for delete broker instance 

### PR DESCRIPTION
**Pull request by Msarawan**
Pre-Confirm Delete Modal Image :

Screenshot from 2023-03-23 18-23-04

### CODE REVIEW

File: src/brokers/view-brokers/BrokersList.container.tsx [hunk 1]
```diff
@@ -17,6 +18,8 @@ const BrokersContainer: FC<BrokersContainerProps> = ({ match }) => {
   const [brokers, setBrokers] = useState<K8sResourceKind[]>();
   const [loading, setLoading] = useState<boolean>(true);
   const [loadError, setLoadError] = useState<any>();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [onDeleteClick, setOnDeleteClick] = useState(null);
```
Comment (ajaypratap003) [inline src/brokers/view-brokers/BrokersList.container.tsx:22]:
```suggestion
  const [selectedBroker, setSelectedBroker] = useState< K8sResourceKind >();
```

File: src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { useTranslation } from '../../../../i18n';
+import { Trans } from 'react-i18next';
+
+interface OnDeleteClickProps {
+  isModalOpen: boolean;
+  onDeleteButtonClick: () => void;
+  handleModalToggle: () => void;
+}
+
+const PreConfirmDeleteModal: FC<OnDeleteClickProps> = ({
```
Comment (ajaypratap003) [inline src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx:12]:
```suggestion
const PreConfirmDeleteModal: FC< PreConfirmDeleteModalProps > = ({
```

File: src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  onDeleteButtonClick,
+  isModalOpen,
+  handleModalToggle,
+}) => {
+  const { t } = useTranslation();
+
+  return (
+    <Fragment>
```
Comment (ajaypratap003) [inline src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx:20]:
```suggestion
    
```
We don't need fragment here. Fragment require when we have more than one sibling tag

File: locales/en/plugin__activemq-artemis-self-provisioning-plugin.json [hunk 4]
```diff
@@ -26,5 +26,9 @@
   "yes": "Yes",
   "no": "No",
   "active": "Active",
-  "inactive": "Inactive"
+  "inactive": "Inactive",
+  "delete_instance_?": "Delete instance ?",
+  "delete_broker_instance": "Delete",
```
Comment (ajaypratap003) [inline locales/en/plugin__activemq-artemis-self-provisioning-plugin.json:31]:
```suggestion
  "delete: "Delete",
```

File: locales/en/plugin__activemq-artemis-self-provisioning-plugin.json [hunk 5]
```diff
@@ -26,5 +26,9 @@
...
+  "cancel_broker_instance": "Cancel",
```
Comment (ajaypratap003) [inline locales/en/plugin__activemq-artemis-self-provisioning-plugin.json:32]:
```suggestion
  "cancel": "Cancel",
```

File: src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      <Modal
+        variant={ModalVariant.small}
+        title={t('delete_instance_?')}
+        titleIconVariant="warning"
+        isOpen={isModalOpen}
+        onClose={handleModalToggle}
+        actions={[
+          <Button key="delete" variant="primary" onClick={onDeleteButtonClick}>
+            {t('delete_broker_instance')}
```
Comment (ajaypratap003) [inline src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx:29]:
```suggestion
            {t('delete')}
```

File: src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+          </Button>,
+          <Button key="cancel" variant="link" onClick={handleModalToggle}>
+            {t('cancel_broker_instance')}
```
Comment (ajaypratap003) [inline src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx:32]:
```suggestion
            {t('cancel')}
```

**Review by ajaypratap003 — CHANGES_REQUESTED** [review]
@Msarawan I've added few suggestions.Please follow inline comments. 
Please attached screenshot of delete modal in PR description.

**Comment by ajaypratap003** [comment]
Please also fix PR build issue.

File: locales/en/plugin__activemq-artemis-self-provisioning-plugin.json [hunk 8]
```diff
@@ -33,5 +33,5 @@
   "delete_instance_?": "Delete instance ?",
   "delete_broker_instance": "Delete",
   "cancel_broker_instance": "Cancel",
-  "preconfirm_delete_broker_message": "The 'test-2' instance will be deleted. Applications will no longer have access in this instance."
+  "preconfirm_delete_broker_message": "The <strong>'test-2'</strong> instance will be deleted. Applications will no longer have access in this instance."
```
[the remark this answers is not in the document]
Comment (Msarawan) [reply locales/en/plugin__activemq-artemis-self-provisioning-plugin.json]: sure Ajay...

File: locales/en/plugin__activemq-artemis-self-provisioning-plugin.json [hunk 9]
```diff
@@ -31,7 +31,7 @@
   "auto_create_queues": "Auto create queues",
   "auto_delete_queues": "Auto delete queues",
   "delete_modal_instance_title": "Delete instance ?",
-  "delete": "Delete",
-  "cancel": "Cancel",
-  "preconfirm_delete_broker_message": "The 'test-2' instance will be deleted. Applications will no longer have access in this instance."
+  "delete": "<strong>Delete</strong>",
```
Comment (ajaypratap003) [inline locales/en/plugin__activemq-artemis-self-provisioning-plugin.json:34]: We don't need <strong> tag for delete and cancel.

File: src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx [hunk 10]
```diff
@@ -25,14 +34,21 @@ const PreConfirmDeleteModal: FC<PreConfirmDeleteModalProps> = ({
       onClose={onOpenModal}
       actions={[
         <Button key="delete" variant="primary" onClick={onDeleteButtonClick}>
-          {t('delete')}
+          <Trans>{t('delete')}</Trans>
```
Comment (ajaypratap003) [inline src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx:37]: We don't need <Trans> here.

File: src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx [hunk 11]
```diff
@@ -25,14 +34,21 @@ const PreConfirmDeleteModal: FC<PreConfirmDeleteModalProps> = ({
...
         </Button>,
         <Button key="cancel" variant="link" onClick={onOpenModal}>
-          {t('cancel')}
+          <Trans>{t('cancel')}</Trans>
```
Comment (ajaypratap003) [inline src/brokers/view-brokers/components/PreConfirmDeleteModal/PreConfirmDeleteModal.tsx:40]: We don't need for cancel

**Review by ajaypratap003 — CHANGES_REQUESTED** [review]
@Msarawan  Added few suggestions. Please follow inline comments. Rest looks good.

File: locales/en/plugin__activemq-artemis-self-provisioning-plugin.json (hunk 9 above)
Comment (Msarawan) [reply locales/en/plugin__activemq-artemis-self-provisioning-plugin.json:34]: okay Ajay, i will reomve the tag i just thought to make delete button to looks  bold so inserted the tag...

**Review by ajaypratap003 — APPROVED** [review]

**Pull request merged**